test/form.test.ts

```diff
@@ -2732,3 +2732,35 @@ test("form with thousandSeparator and decimalSeparator same value invalid", asyn
     });
   }).toThrow();
 });
+
+test("setValueAndUpdateRaw", async () => {
+  const M = types.model("M", {
+    foo: types.string
+  });
+
+  const o = M.create({ foo: "" });
+
+  const form = new Form(M, {
+    foo: new Field(converters.decimal())
+  });
+
+  const state = form.state(o, {
+    converterOptions: {
+      thousandSeparator: ".",
+      decimalSeparator: ",",
+      renderThousands: true
+    }
+  });
+
+  // Setting the raw directly would update the value without relying on other event handlers
+  const field = state.field("foo");
+
+  await field.setRaw("1234.56");
+  expect(field.raw).toEqual("1234.56");
+  expect(field.value).toEqual("1234.56");
+
+  // Instead, we set the value and update the raw based on the value
+  await field.setValueAndUpdateRaw("1234.56");
+  expect(field.raw).toEqual("1.234,56");
+  expect(field.value).toEqual("1234.56");
```

Comment:
What I meant with my last comment (and what I now see again) is that I can't tell if the setValue part of setValueAndUpdateRaw even does anything. I know it's obvious that it does, given its definition, but it's not reflected in the test. field.value is already 1234.56, and after setValueAndUpdateRaw, it's _still_ 1234.56. Has setValueAndUpdateRaw set the value? I know that it has, based on its definition, but this test doesn't showcase that, because this is the same value.